src/main/scala/com/picnicml/doddlemodel/base/Clusterer.scala

```diff
@@ -0,0 +1,20 @@
+package com.picnicml.doddlemodel.base
+
+import java.io.Serializable
+
+import com.picnicml.doddlemodel.data.{Features, Target}
+
+abstract class Clusterer[A <: Clusterer[A]] extends Estimator {
```

Comment:
`Clusterer` currently only has a single publicly exposed function: `def predict(x: Features): Target`. Additionally, I don't think all clustering algorithms will expose `predict`, e.g. DBSCAN doesn't. On the other hand, all of them probably need `fit` exposed?